torch/distributed/tensor/debug/visualize_sharding_example.py

```diff
@@ -0,0 +1,77 @@
+# OMP_NUM_THREADS=1 torchrun --nproc_per_node=4 visualize_sharding_example.py
+import contextlib
+import importlib.util
+import os
+
+import numpy as np
+import torch
+import torch.distributed as dist
+import torch.distributed.elastic.multiprocessing.errors
+import torch.distributed.tensor as dt
+from torch._prims_common import ShapeType
+
+from _visualize_sharding import visualize_sharding
+
+
+@contextlib.contextmanager
+def distributed_context():
+    try:
+        local_rank = int(os.environ["LOCAL_RANK"])
+        local_device = torch.device("cpu", local_rank)
+        dist.init_process_group(backend="gloo")
+        yield local_device
+    finally:
+        dist.barrier()
+        dist.destroy_process_group()
+        print(f"Rank {local_rank} finished")
+
+
+@torch.distributed.elastic.multiprocessing.errors.record
+def main(local_device):
+    local_rank = local_device.index
+    tensor = torch.ones((2, 2), device=local_device) * local_rank
+
+    # Case 1: 1D mesh
+    mesh = dist.init_device_mesh("cpu", (4,), mesh_dim_names=["dp"])
+    visualize_sharding(dt.DTensor.from_local(tensor, mesh, [dt.Replicate()]))
+    visualize_sharding(dt.DTensor.from_local(tensor, mesh, [dt.Replicate()]), use_rich=True)
+    visualize_sharding(dt.DTensor.from_local(tensor, mesh, [dt.Shard(dim=0)]))
+    visualize_sharding(dt.DTensor.from_local(tensor, mesh, [dt.Shard(dim=0)]), use_rich=True)
+    visualize_sharding(dt.DTensor.from_local(tensor, mesh, [dt.Shard(dim=1)]))
+    visualize_sharding(dt.DTensor.from_local(tensor, mesh, [dt.Shard(dim=1)]), use_rich=True)
+
+    # Case 2: 2D mesh
+    mesh = dist.init_device_mesh("cpu", (2, 2), mesh_dim_names=["dp", "tp"])
```

Comment:
hmm. this example script is hanging for me.  I noticed that py-spy reports one rank is stuck on L24 barrier(), another rank is stuck on L44 init_device_mesh.

I also noticed that even though there were many calls to 'visualize_sharding' I only actually see one tabulate and one rich table.  
<img width="980" alt="image" src="https://github.com/user-attachments/assets/9bb1f756-c1e7-4dc8-8dad-b157307a0026" />

Reply:
I haven't been able to run it as part of PyTorch. I am still building PyTorch from source code on a Linux computer. I gave up building it on macOS.

Could you try:

```
pip install dtensor
torchrun --nproc_per_node=4 $(pip show dtensor | grep Location: | cut -f 2 -d ' ')/dtensor/[visualize_example.py](http://visualize_example.py/)
```

Here follows what I get. I don't see log message like `[Gloo] Rank 0 is connected to 3 peer ranks` as they are in your screenshot.

<img width="476" alt="Screenshot 2025-04-23 at 9 29 38 PM" src="https://github.com/user-attachments/assets/574d4eaf-c278-4137-bc0e-dd91bf7429aa" />